### DISCUSSION TITLE
Only need to copy airgap tarball images to first server when using embedded registry?

### DISCUSSION THREAD

**Discussion by matthewadams** [k3s-io/k3s#10052] [Q&A] [answered]
https://docs.k3s.io/installation/airgap#embedded-registry-mirror says:

> The mirrored images may be sourced from an upstream registry, registry mirror, or airgap image tarball.

I just want to clarify.  This seems to imply to me that I can import the airgap tarball images into the embedded registry on the first server, and any subsequent server install will be able to pull those airgap images from the embedded registry, such that I don't have to copy the airgap image tarball to each subsequent server node.

EDIT:
I just realized that I need to ask this question of not only the airgap images tarball, but also of all images listed in each release's `https://github.com/k3s-io/k3s/releases/download/v$VERSION/k3s-images.txt`.
END EDIT

Is this correct?

**Comment by brandond** [comment, marked as answer]
> I can import the airgap tarball images into the embedded registry on the first server, and any subsequent server install will be able to pull those airgap images from the embedded registry, such that I don't have to copy the airgap image tarball to each subsequent server node.

That is correct. As long as that initial node is available to "seed" the images to other nodes, you don't need to put them anywhere else.

This isn't just restricted to just our tarballs, you can put your own image tarballs as produced by `docker save` or other compatible tools. 

Other pages of the documentation cover this in more detail: https://docs.k3s.io/installation/registry-mirror#sharing-air-gap-or-manually-loaded-images

**Reply by matthewadams** [reply]
Thanks! 🙏️
